integrationtests/Paket.IntegrationTests/PackSpecs.fs

```diff
@@ -586,11 +586,6 @@ let ``#2694 paket fixnuspec should not remove project references``() =
     | None -> Assert.Fail("Expected package to still contain the FSharp.Core reference!")
     | Some s -> ignore s
 
-    // Should we remove Microsoft.NETCore.App?
```

Comment:
@matthid sorry to bother you mate. This is the only integration test that is red. I tried to port everything to the SDK and somewhere here in this PR there is something that changes nuspec.Dependencies.Value.Length from 3 to 2.
The comment above the assertion is already very suspicious, but still I'm wondering what changed. Any ideas?

Reply:
Can you you check and post what was in the list before and what is there now to see what has disappeared?

Github seems to make it a bit harder to see the code when force pushes are used apparently...

Reply:
Dependencies: 

    [(PackageName ("library","library"), VersionRequirement (Minimum 1.0.0,No),
      ExplicitRestriction >= netcoreapp2.0);
     (PackageName ("FSharp.Core","fsharp.core"),
      VersionRequirement (Minimum 4.2.3,No), ExplicitRestriction >= netcoreapp2.0);
     (PackageName ("Microsoft.NETCore.App","microsoft.netcore.app"),
      VersionRequirement (Minimum 2.0.0,No), ExplicitRestriction >= netcoreapp2.0)]

so it's removing Microsoft.NETCore.App

Reply:
I think Microsoft.NETCore.App is indeed special in a way. 
Question is if we have added some kind of special casing for it in the code-base?
Seems like it was magically added before this PR, because in the lockfile of this test it is not even referenced, see https://github.com/fsprojects/Paket/blob/fullsdk/integrationtests/scenarios/i002694/before/paket.lock
So I guess it is "safe" that it is no longer there. I'm more wondering why it was there before this PR, which only really changes project setup?

Maybe some DEFINES are not like they where before?

Reply:
Yes. I tried convert the compiler switches but maybe I lost one.
I also updated the sdk to 3.0.100. Maybe that's relevant?

Reply:
ok confirmed. I updated the sdk in the test PR (see https://github.com/fsprojects/Paket/pull/3718) and the dependency is gone as well.

Reply:
Ah because we use the SDK nuspec output as input for ourself. It seems like they have decided to filter this one. So at least that confirms that we should filter as well if needed?